zone/spells.cpp

```diff
@@ -4187,41 +4187,39 @@ bool Mob::SpellOnTarget(
 		spelltar->DamageShield(this, true);
 	}
 
-	if (
-		spelltar->IsAIControlled() &&
-		IsDetrimentalSpell(spell_id) &&
-		!IsHarmonySpell(spell_id)
-	) {
-		auto aggro_amount = CheckAggroAmount(spell_id, spelltar, isproc);
-		LogSpells("Spell [{}] cast on [{}] generated [{}] hate", spell_id,
-			spelltar->GetName(), aggro_amount);
-		if (aggro_amount > 0) {
-			spelltar->AddToHateList(this, aggro_amount);
-		} else {
-			int64 newhate = spelltar->GetHateAmount(this) + aggro_amount;
-			spelltar->SetHateAmountOnEnt(this, std::max(newhate, static_cast<int64>(1)));
-		}
-	} else if (IsBeneficialSpell(spell_id) && !IsSummonPCSpell(spell_id)) {
-		if (this != spelltar && IsClient()){
-			if (spelltar->IsClient()) {
-				CastToClient()->UpdateRestTimer(spelltar->CastToClient()->GetRestTimer());
-			} else if (spelltar->IsPet()) {
-				auto* owner = spelltar->GetOwner();
-				if (owner && owner != this && owner->IsClient()) {
-					CastToClient()->UpdateRestTimer(owner->CastToClient()->GetRestTimer());
+	if (!(spelltar->IsNPC() && IsNPC() && !(IsPet() && GetOwner()->IsClient()))) {
```

Comment:
If I'm understanding this right this is preventing hate being added if an npc (that's not a client charmed pet) casts on another npc? The downside for forced npc vs npc fights is healers or caster npcs won't accumulate hate but guessing this is the main cause of unintentional faction wars?

Reply:
Correct. inherent npc to npc aggro (Conflicting Factions) should still continue to work but will not accumulate hate. In my limited tested years ago, the mobs would always target a real player on hate over a npc, so this didn't seem like an issue with this.

We just want to ensure that npc will not accidentally get onto the hatelist of another "friendly" or "indifferent" faction.

Prob not the most elegant way to prevent it, but it has worked well for me.